Framework/CurveFitting/inc/MantidCurveFitting/ComplexVector.h

```diff
@@ -0,0 +1,153 @@
+#ifndef MANTID_CURVEFITTING_COMPLEXVECTOR_H_
+#define MANTID_CURVEFITTING_COMPLEXVECTOR_H_
+
+#include "MantidCurveFitting/DllConfig.h"
+#include <gsl/gsl_vector.h>
+
+#include <ostream>
+#include <vector>
+#include <complex>
+
+namespace Mantid {
+namespace CurveFitting {
+
+typedef std::complex<double> ComplexType;
+class ComplexVector;
+
+/// Struct helping converting complex values
+/// between ComplexType and internal type of
+/// ComplexVector
+struct ComplexVectorValueConverter {
+  ComplexVector &m_vector;
+  size_t m_index;
+  ComplexVectorValueConverter(ComplexVector &vector, size_t i)
+      : m_vector(vector), m_index(i) {}
+  operator ComplexType() const;
+  ComplexVectorValueConverter &operator=(const ComplexType &c);
+};
+
+/**
+A complex-valued vector for linear algebra computations.
+
+Copyright &copy; 2010 ISIS Rutherford Appleton Laboratory, NScD Oak Ridge
+National Laboratory & European Spallation Source
+
+This file is part of Mantid.
+
+Mantid is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 3 of the License, or
+(at your option) any later version.
+
+Mantid is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+File change history is stored at: <https://github.com/mantidproject/mantid>
+Code Documentation is available at: <http://doxygen.mantidproject.org>
+*/
+class MANTID_CURVEFITTING_DLL ComplexVector {
+public:
+  /// Constructor
+  ComplexVector();
+  /// Destructor
+  ~ComplexVector();
+  /// Constructor
+  explicit ComplexVector(const size_t n);
+  /// Copy from a gsl vector
+  explicit ComplexVector(const gsl_vector_complex *v);
+  /// Copy constructor.
+  ComplexVector(const ComplexVector &v);
```

Comment:
We should probably have a move constructor & move assignment operator here too. They would have been suppressed by the user-defined destructor, copy-assign & copy constructor.